linkerd/protocol/http/src/main/scala/com/twitter/finagle/buoyant/linkerd/Headers.scala

```diff
@@ -33,21 +33,21 @@ object Headers {
         traceId <- get(header).toOption
       } yield traceId
 
-    def set(headers: HeaderMap, id: TraceId): Unit = {
+    def set(headers: HeaderMap, id: TraceId): HeaderMap = {
       val bytes = TraceId.serialize(id)
       val b64 = Base64.getEncoder().encodeToString(bytes)
       headers.set(Key, b64)
     }
 
-    def clear(headers: HeaderMap): Unit = {
+    def clear(headers: HeaderMap): Option[String] = {
       headers.remove(Key)
     }
```

Comment:
looks like you changed the other set of set/clear functions, but these should still be `Unit`...